test/e2e/nodelatencymonitor_test.go

```diff
@@ -0,0 +1,124 @@
+		if len(statsList.Items) != clusterInfo.numNodes {
+			t.Logf("Expected %d NodeLatencyStats, but found %d, retrying...", clusterInfo.numNodes, len(statsList.Items))
+			return false, nil
+		}
+
+		for _, item := range statsList.Items {
+			if len(item.PeerNodeLatencyStats) != clusterInfo.numNodes-1 {
+				t.Logf("Expected %d PeerNodeLatencyStats for Node %s, but found %d, retrying...", clusterInfo.numNodes-1, item.Name, len(item.PeerNodeLatencyStats))
+				return false, nil
+			}
+
+			for _, peerStat := range item.PeerNodeLatencyStats {
+				if len(peerStat.TargetIPLatencyStats) != expectedTargetIPLatencyStats {
+					t.Logf("Expected %d TargetIPLatencyStats for peer %s on Node %s, but found %d, retrying...",
+						expectedTargetIPLatencyStats, peerStat.NodeName, item.Name, len(peerStat.TargetIPLatencyStats))
+					return false, nil
+				}
+
+				for _, targetStat := range peerStat.TargetIPLatencyStats {
+					if targetStat.LastMeasuredRTTNanoseconds <= 0 {
+						t.Logf("Invalid RTT for peer %s reported by Node %s", peerStat.NodeName, item.Name)
+						return false, nil
+					}
+
+					if initialPoll {
+						if previousTimes[item.Name] == nil {
+							previousTimes[item.Name] = make(map[string]metav1.Time)
+						}
+						previousTimes[item.Name][peerStat.NodeName] = targetStat.LastRecvTime
+					} else {
+						previousRecvTime, recvTimeExists := previousTimes[item.Name][peerStat.NodeName]
+						if !recvTimeExists || !targetStat.LastRecvTime.After(previousRecvTime.Time) {
+							t.Logf("LastRecvTime has not been updated for peer %s on Node %s", peerStat.NodeName, item.Name)
+							return false, nil
+						}
+					}
+				}
+			}
+		}
+		return true, nil
+	}
+
+	err = wait.PollImmediate(time.Second, 30*time.Second, func() (bool, error) {
+		statsList, err := data.crdClient.StatsV1alpha1().NodeLatencyStats().List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		valid, validateErr := validateNodeLatencyStats(statsList, true)
+		if !valid {
+			return false, validateErr
+		}
+		return true, nil
+	})
+	require.NoError(t, err, "Failed to retrieve initial NodeLatencyStats")
+
+	err = wait.PollImmediate(time.Second, 30*time.Second, func() (bool, error) {
+		statsList, err := data.crdClient.StatsV1alpha1().NodeLatencyStats().List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		valid, validateErr := validateNodeLatencyStats(statsList, false)
+		if !valid {
+			return false, validateErr
+		}
+		return true, nil
+	})
+	require.NoError(t, err, "Failed to validate NodeLatencyStats after update")
+
+	t.Logf("Successfully received and validated NodeLatencyStats")
+
+	err = data.crdClient.CrdV1alpha1().NodeLatencyMonitors().Delete(context.TODO(), "default", metav1.DeleteOptions{})
+	require.NoError(t, err, "Failed to delete NodeLatencyMonitor CR")
+	t.Logf("NodeLatencyMonitor CR deleted successfully.")
```

Comment:
I just realized that this should be a `defer` statement just after the `Create` operation succeeds. Otherwise the test may fail (e.g., during one of the poll) and then the CR will not be deleted and will persist after the test case.

Reply:
yes  , you are right .
I'm changing.